### ISSUE TITLE
api.proto - versioning & single source of truth

### ISSUE THREAD

**Issue by rdewaele** [dgraph-io/dgo#16]
Hello,

I hope this is the correct channel to post my question. If it is not, I apologise and kindly request redirection to the appropriate location. Thanks! On to the actual question:

I have a basic (functional) custom C++ gRPC client for Dgraph, based on the [protos/api.proto](https://github.com/dgraph-io/dgo/blob/1e7766a7c63a03daf86faf59524df1bfa4ecc4d9/protos/api.proto) file in this repository, as it is linked by the [Dgraph client documentation](https://docs.dgraph.io/clients/).

Given that a number of different official clients exist, and that Dgraph itself has a number of releases (branches, tags, archives), but this client does not, I wonder about gRPC compatibility.

In other words, does this repository host the ultimate "correct" version of the api.proto file, and how should we proceed to maintain compatibility with Dgraph current and future releases?

Thanks!

**Comment by srfrog** [comment]
Yes, this has the final API definition. The server code uses this proto so they must be kept in sync.
I would suggest you base your client from the api.proto in here.

Please post these questions in the forum, so others can benefit - https://discuss.dgraph.io
Thanks.
